File: googleq.py
#given a sorted array and a sum, return whether the array contains a pair
#equals the sum
sarr = [0, 2, 13, 23, 39, 56, 90, 103]
def google(sarr, sum):
    start = 0
    end = len(sarr)-1
    while start < end:
        total = sarr[start] + sarr[end]
        if total == sum:
            return True
        if total < sum:
            start = start + 1
        if total > sum:
            end = end - 1
    return False
print(google(sarr, 48))

File: googleq.js
//given a sorted array and a sum, return whether the array contains a pair
//equals the sum

const sarr = [0, 2, 13, 23, 39, 56, 90, 103]
function google(sarr, sum){
    let start = 0
    let end = sarr.length-1
    let result = false;
    while(start < end){
        let total = sarr[start] + sarr[end];
        if(total === sum){
            result = true;
            break;
        }
        if(total < sum){
            start++;
        }
        if(total > sum){
            end--;
        }
    }
    return result;
}
console.log("Google Q:", google(sarr, 13));